dynamiq/nodes/tools/basetool.py

```diff
@@ -0,0 +1,45 @@
+from abc import ABC, abstractmethod
+from typing import Any, Literal
+
+from pydantic import BaseModel, Field, ValidationInfo, field_validator
+
+from dynamiq.nodes import NodeGroup
+from dynamiq.runnables import RunnableConfig
+
+
+class DefaultInputSchema(BaseModel, ABC):
+    input: str
+
+
+class ToolMixin:
+    group: Literal[NodeGroup.TOOLS] = NodeGroup.TOOLS
+    name: str = "base_tool"
+    description: str = "base tool"
+    input_schema: type[BaseModel] = Field(default=DefaultInputSchema)
+
+    @field_validator("input_schema")
+    def passwords_match(cls, v: BaseModel, info: ValidationInfo) -> BaseModel:
+        # Check if input_schema is correct
+        return v
+
+    @abstractmethod
+    def run_tool(self):
+        pass
+
+    @property
+    def input_params(self) -> str:
```

Comment:
As schema on class level, this helper also should be classmethod

Reply:
Class properties are deprecated, so I added separate `get_input_schema` method